.github/workflows/cmake_build.yml

```diff
@@ -0,0 +1,259 @@
+name: Build and Test on Push
+on:
+  push:
+  pull_request:
+    branches: [master]
+
+jobs:
+  # These test the 'standard' build on several systems with gcc + clang.
+  standard_tests:
+    name: ${{ matrix.os }} & ${{ matrix.compiler.CC }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-16.04, ubuntu-18.04, 	ubuntu-20.04, macos-10.15]
+        compiler:
+          - { CC: gcc, CXX: g++ }
+          - { CC: clang, CXX: clang++ }
+
+    steps:
+      - name: Get Source
+        uses: actions/checkout@v2
+  
+      - name: "Linux: Run Apt-Get"
+        if: startsWith(matrix.os, 'ubuntu') && matrix.os != 'ubuntu-20.04'
+        run: sudo apt-get install elfutils libzstd1-dev
+  
+      - name: "Linux (Ubuntu 20.04): Run Apt-Get"
+        if: matrix.os == 'ubuntu-20.04'
+        run: sudo apt-get install elfutils libzstd-dev
+  
+      - name: "Setup common ENV"
+        uses: allenevans/set-env@v1.1.0
+        with:
+          CC: ${{ matrix.compiler.CC }}
+          CXX: ${{ matrix.compiler.CXX }}
+          VERBOSE: 1
+          ENABLE_CACHE_CLEANUP_TESTS: true
+          CTEST_OUTPUT_ON_FAILURE: ON
+      
+      - name: Build and Test
+        shell: bash
+        run: ci/build
+  
+      - name: Collect testdirs from failed tests
+        if: failure()
+        run: zip -r testdirs.zip ${{ matrix.config.BUILDDIR }}/testdir.* -x testdir.failed
+
+      - name: Upload testdirs from failed tests
+        if: failure()
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.config.name }} - testdirs.zip
+          path: testdirs.zip
+
+      # Does it make sense to do this on every matrix element?
+      - name: "Build binary and verify it"
+        run: ci/build-and-verify-package
+        # ToDo: upload testdirs in case of errors
+
+      - name: Upload binary
+        uses: actions/upload-artifact@v2
+        with:
+          # Does it make sense to upload all of them?
+          # What is even the use case? People using specific versions without compiling on their own.
```

Comment:
I don't think that it makes sense to upload any Linux binaries, actually, unless they are statically linked (except possibly to libc). And if we do they should be a release build.

Reply:
They are statically linked, but not release. I'll remove it for now.